src/LightStep/Collector/LightStepHttpClient.cs

```diff
@@ -34,25 +36,41 @@ public LightStepHttpClient(string url, Options options)
         /// </summary>
         /// <param name="report">An <see cref="ReportRequest" /></param>
         /// <returns>A <see cref="ReportResponse" />. This is usually not very interesting.</returns>
-        public ReportResponse SendReport(ReportRequest report)
+        public async Task<ReportResponse> SendReport(ReportRequest report)
         {
             // force net45 to attempt tls12 first and fallback appropriately
             ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls12 | SecurityProtocolType.Tls11 | SecurityProtocolType.Tls;
             
             _client.DefaultRequestHeaders.Accept.Add(
                 new MediaTypeWithQualityHeaderValue("application/octet-stream"));
+            var reportsByteArray = report.ToByteArray();
 
             var request = new HttpRequestMessage(HttpMethod.Post, _url)
             {
                 Version = _options.UseHttp2 ? new Version(2, 0) : new Version(1, 1),
-                Content = new ByteArrayContent(report.ToByteArray())
+                Content = new ByteArrayContent(reportsByteArray)
             };
 
             request.Content.Headers.ContentType = new MediaTypeHeaderValue("application/octet-stream");
+            
+            ReportResponse responseValue = new ReportResponse();
+            
+            try
+            {
+                var response = await _client.SendAsync(request);
+                response.EnsureSuccessStatusCode();
+                var responseData = await response.Content.ReadAsStreamAsync();
+                responseValue = ReportResponse.Parser.ParseFrom(responseData);
+            }
+            catch (HttpRequestException ex)
+            {
+                Console.WriteLine(ex);
```

Comment:
Assuming the exception info includes a status code

Reply:
Yeah, it does.